src/Types/Infer/Class.hs

```diff
@@ -433,20 +445,20 @@ reduceClassContext extra annot cons = do
   needs :: [Need Typed] <- sortOn (view _1) <$> needed mempty cons
 
   -- First, deduplicate the constraints eliminating any redundancy
-  let dedup :: ImplicitScope Typed -> [Need Typed] -> ([Binding Typed], [Need Typed], ImplicitScope Typed)
+  let dedup :: ImplicitScope ClassInfo Typed -> [Need Typed] -> ([Binding Typed], [Need Typed], ImplicitScope ClassInfo Typed)
       dedup scope ((var, con, r):needs)
-        | [ImplChoice _ t [] v _ _] <- lookup con scope =
+        | [ImplChoice _ t [] v _ _ _] <- lookup con scope =
           let (bindings, needs', scope') = dedup scope needs
            in if var == v
              then (bindings, needs', scope')
              else (Binding var (VarRef v (annot, t)) False (annot, t):bindings, needs', scope')
         | otherwise =
-          let (bindings, needs', scope') = dedup (insert annot LocalAssum var con scope) needs
+          let (bindings, needs', scope') = dedup (insert annot LocalAssum var con undefined scope) needs
```

Comment:
Might be worth commenting why this can be `undefined`?

Reply:
aye

Reply:
donezoe